crates/iceberg/src/static_table.rs

```diff
@@ -0,0 +1,146 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! `StaticTable` is a read-only table struct that can be created from a metadata file or from `TableMetaData` without a catalog.
+//! It can only be used to read metadata and for table scan.
+//! For example:
+//! ```rust, compile_fail
```

Comment:
Why this compile failed? I think the compilation can pass by adding some `using` statement?

Reply:
I added the `compile_fail` flag because the example involves async code that requires runtime, and I didn't want to mess up the code example. I will add it.